pkg/chattr/chattr.go

```diff
@@ -16,34 +16,74 @@ const (
 	lsattrCmd = "lsattr"
 )
 
-func AddImmutable(path string) error {
+func AddImmutable(path string) (err error) {
+
+	return ForceAddImmutable(path, false)
+}
+
+func ForceAddImmutable(path string, force bool) (err error) {
 	chattrBin := which(chattrCmd)
-	if _, err := os.Stat(path); err == nil {
-		cmd := exec.Command(chattrBin, "+i", path)
+	if _, err = os.Stat(path); err == nil {
+		var cmd *exec.Cmd
+		if force {
+			cmd = exec.Command(chattrBin, "+i", "-f", path)
+		} else {
+			cmd = exec.Command(chattrBin, "+i", path)
+		}
 		var stderr bytes.Buffer
 		cmd.Stderr = &stderr
 
-		if err = cmd.Run(); err != nil {
+		err = cmd.Run()
+		if err != nil {
+			// chattr exits with status 1 for files that are not permitted without force flag
+			// this will happen for files that throws the error
+			//  "Operation not supported while reading flags"
+			if force && cmd.ProcessState.ExitCode() == 1 {
+				return nil
+			}
+			if force {
+				return fmt.Errorf("%s +i -f failed: %s. Err: %v", chattrBin, stderr.String(), err)
+			}
 			return fmt.Errorf("%s +i failed: %s. Err: %v", chattrBin, stderr.String(), err)
 		}
+		return nil
 	}
 
-	return nil
+	return err
 }
 
-func RemoveImmutable(path string) error {
+func RemoveImmutable(path string) (err error) {
+	return ForceRemoveImmutable(path, false)
+}
+func ForceRemoveImmutable(path string, force bool) (err error) {
```

Comment:
Same comments as above for `ForceAddImmutable` as the functions are essentially the same with `+i/-i`